rust/headless-client/README.md

```diff
@@ -0,0 +1,65 @@
+# headless-client
+
+This crate acts as the CLI / headless Client, and the privileged tunnel service for the GUI Client, for both Linux and Windows.
+
+It is built as:
+- `linux-client` to act as the Linux headless Client
```

Comment:
Why do we still need this?

Reply:
I figured the filename is part of the API, I don't know which customers have the Linux Client installed and will have a problem if we need to change the name. @jamilbk probably knows

Reply:
Let's bias towards changing it now while the impact is low if it prevents tech debt.

If you change this binary, could you grep for it in `website/src/app/kb` and change it there too? I think we link directly to it from the `user-guides/linux-client` section

Reply:
Actually, don't do that because that change needs to happen when we publish the next release. I'll note it in #3988

Reply:
Okay so I'll still try to keep all user-visible names the same - `linux-client` in Github releases, etc. It'll just be the Rust packages shifting around, and we'll make a breaking change later to align the public API with the internal structure.

Reply:
> Actually, don't do that because that change needs to happen when we publish the next release. I'll note it in #3988

This was already merged now, so what does that mean? Can we have a follow-up issue for the actual rename?

Reply:
https://github.com/firezone/firezone/issues/4531